pythoncode/FortiusAntBody.py

```diff
@@ -666,15 +646,27 @@ def Tacx2Dongle(self):
     hrm.Initialize()
     fe.Initialize()
     
+    #---------------------------------------------------------------------------
+    # Initialize CycleTime: fast for PedalStrokeAnalysis
+    #---------------------------------------------------------------------------
     if clv.PedalStrokeAnalysis:
         CycleTime = CycleTimeFast   # Quick poll to get more info
         if debug.on(debug.Any): logfile.Console("Tacx2Dongle; Pedal Stroke Analysis active")
     else:
         CycleTime = CycleTimeANT    # Seconds, default = 0.25 (inspired by 4Hz ANT+)
 
+    #---------------------------------------------------------------------------
+    # Our main loop!
+    # The loop has the following phases
+    # -- Get data from trainer
+    # -- Local adjustments (heartrate monitor, cadence sensor)
+    # -- Display actual values
+    # -- Pedal stroke analysis
+    # -- Modify data, due to Buttons or ANT
+    #---------------------------------------------------------------------------
     if debug.on(debug.Function): logfile.Write('Tacx2Dongle; start main loop')
     try:
-        while self.RunningSwitch == True:
+        while self.RunningSwitch == True and not AntDongle.DongleReconnected:

```

Comment:
Traceback (most recent call last):
  File "FortiusAnt.py", line 520, in <module>
    Console.Autostart()
  File "FortiusAnt.py", line 161, in Autostart
    Tacx2Dongle(self)
  File "FortiusAnt.py", line 110, in Tacx2Dongle
    rtn = FortiusAntBody.Tacx2Dongle(self)
  File "/home/pi/WouterJD/FortiusANT/pythoncode/FortiusAntBody.py", line 448, in Tacx2Dongle
    rtn = Tacx2DongleSub(self, Restart)
  File "/home/pi/WouterJD/FortiusANT/pythoncode/FortiusAntBody.py", line 669, in Tacx2DongleSub
    while self.RunningSwitch == True and not AntDongle.DongleReconnected:
AttributeError: 'NoneType' object has no attribute 'DongleReconnected'

Reply:
Yep. Already found it and solved